homeassistant/components/legrand_rflc/light.py

```diff
@@ -0,0 +1,180 @@
+        )
+
+    @staticmethod
+    def _from_ha(value) -> int:
+        return _LegrandRflcDimmer._quantize(
+            _LegrandRflcDimmer._normalize(value, _LegrandRflcDimmer.HA),
+            _LegrandRflcDimmer.US,
+        )
+
+    def __init__(self, hub, zid: int, properties: Mapping):
+        super().__init__(hub, zid, properties)
+        self._attr_brightness = self._to_ha(properties[hub.POWER_LEVEL])
+
+    async def _zone_properties_changed(self, message: Mapping) -> None:
+        self._zone_properties_changed_switch(message)
+        hub = self._hub
+        if hub.PROPERTY_LIST in message:
+            properties = message[hub.PROPERTY_LIST]
+            if hub.POWER_LEVEL in properties:
+                self._attr_brightness = self._to_ha(properties[hub.POWER_LEVEL])
+        self.async_write_ha_state()
+
+    async def _async_dimmer(self, power: bool, **kwargs) -> None:
+        hub = self._hub
+
+        async def handle(message: Mapping) -> None:
+            hub.StatusError(message).raise_if()
+
+        properties: dict = {"power": power}
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = self._from_ha(kwargs[ATTR_BRIGHTNESS])
+            properties["power_level"] = brightness
+        else:
+            if power:
+                brightness = self._from_ha(self.brightness)
+            else:
+                brightness = 0
+        if ATTR_TRANSITION in kwargs:
+            change = abs(brightness - self._from_ha(self.brightness))
+            properties["ramp_rate"] = min(
+                max(int(change / kwargs[ATTR_TRANSITION]), 1), 100
+            )
+        await hub.handle_send(
+            handle,
+            hub.compose_set_zone_properties(self._zid, **properties),
+        )
+
+    async def async_turn_on(self, **kwargs) -> None:
+        await self._async_dimmer(True, **kwargs)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        await self._async_dimmer(False, **kwargs)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities,
+) -> None:
+    """Set up the Legrand RFLC integration light platform.
+
+    Adds a Dimmer or Switch for each like zone on this entry's hub.
+    """
+    hub = hass.data[DOMAIN][entry.entry_id]
+
+    async def zones(message: Mapping) -> None:
+        async def zone(message: Mapping) -> None:
+            zid = message[hub.ZID]
+            properties = message[hub.PROPERTY_LIST]
+            device_type = properties[hub.DEVICE_TYPE]
+            if device_type == hub.DIMMER:
+                async_add_entities([_LegrandRflcDimmer(hub, zid, properties)], False)
+            elif device_type == hub.SWITCH:
+                async_add_entities([_LegrandRflcSwitch(hub, zid, properties)], False)
+
+        hub.StatusError(message).raise_if()
+        for item in message[hub.ZONE_LIST]:
+            await hub.handle_send(
+                zone, hub.compose_report_zone_properties(item[hub.ZID])
```

Comment:
Does this trigger an initial state update for the lights?

Reply:
yes